api/helpers/gamestate/convert-card-to-Id.js

```diff
@@ -0,0 +1,92 @@
+const cardMap = require('../../temp/cardMap');
+
+/**
+ * Convert a card object to its string representation
+ *
+ * @param {Card} card - card object
+ * @returns {string} String representation of the card
+ */
+function convertCardToId(card){
+  const suitKey =  Object.keys(cardMap.suitMap).find(key => cardMap.suitMap[key] === card.suit);
+
+  const rankKey = card.rank === 1 || card.rank > 9 ? 
+                          Object.keys(cardMap.rankMap).find(key => cardMap.rankMap[key] === card.rank)
+                          : parseInt(card.rank) ;
+
+  if(suitKey === undefined || suitKey === null ){
+    throw new Error('Unrecognisabled suit ' + card.suit ) ;
+  }
+  if(rankKey === undefined || rankKey === null){
+    throw new Error('Unrecognisabled rank ' +  card.rank) ;
+  }
+  return rankKey + suitKey;
+}
+
+ /**
+ * Convert the attachments attribute of a card object to its string representation
+ *
+ * @param {Card, playerid } card, player - Card Object, playerId
+ * @returns {string} string representation of the card atatchements attributs
+ */
+ function concatAttachtToId(card, playedBy){
+    if( card.attachments === null || card.attachments === undefined || card.attachments.length === 0){
+        return '';
+    }
+
+    let playerNum = playedBy === 'p0' ? 'p0' : 'p1'; 
+
+    let attachmentStr ='(';
+
+    //Format => (JH-p0,JC-p1,JD-p0) cardInString + '-' + player (+',')
+    card.attachments.forEach((element, index)=>{
+      const cardInString = convertCardToId( element );
+      let comma = index < card.attachments.length -1 ? ',': '';
+      attachmentStr += cardInString + '-' + playerNum + comma;
+      playerNum = playerNum === 'p0' ? 'p1' : 'p0';
+    });
+
+    return attachmentStr + ')';
```

Comment:
I think this is slightly hard to read, and it might be better to let JS handle adding the commas. Something like this may work.

```suggestion
const convertedCards = card.attachments.map((card) => {
  const cardInString = convertCardToId(card) + '-' + playerNum
  playerNum = playerNum === 'p0' ? 'p1' : 'p0';
  return cardInString
});

return "(" + convertedCards.join(',') + ")";
```